### PR TITLE
fix(sec): avoid sql injection in downtimes page

### PR DESCRIPTION
**Pull request by kduret**
## Description

sanitize request parameters in downtimes page to avoid sql injection

**Fixes** MON-5913

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x
- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [ ] 20.10.x (master)

### CODE REVIEW

File: www/include/monitoring/downtime/listDowntime.php [hunk 1]
```diff
@@ -179,18 +177,25 @@
...
-
-$rows = $pearDBO->numberRows();
-for ($i = 0; $data = $DBRESULT->fetchRow(); $i++) {
+$downtimesStatement = $pearDBO->prepare($request);
+$dbResult = $pearDBO->execute(
+    $stmt,
+    [
+        '%' . $search_service . '%', '%' . $host_name . '%', '%' . $search_output . '%', '%' . $search_author . '%',
+        '%' . $host_name . '%', '%' . $search_output . '%', '%' . $search_author . '%'
+    ]
```
Comment (sc979) [inline www/include/monitoring/downtime/listDowntime.php:194]: isn't a value to bound missing ? (author)

File: www/include/monitoring/downtime/listDowntime.php [hunk 2]
```diff
@@ -179,18 +177,25 @@
...
-    (isset($search_output) && $search_output != "" ? "AND d.comment_data LIKE '%$search_output%' " : "") .
+    "AND h.name LIKE ? " .
+    "AND d.comment_data LIKE ? " .
     (isset($view_all) && $view_all == 0 ? "AND d.end_time > '" . time() . "' " : "") .
     (isset($view_downtime_cycle) && $view_downtime_cycle == 0 ?
         " AND d.comment_data NOT LIKE '%Downtime cycle%' " : "") .
-    (isset($search_author) && $search_author != "" ? " AND d.author LIKE '%$search_author%'" : "") .
+    " AND d.author LIKE ?" .
     ") ORDER BY scheduled_start_time DESC " .
     "LIMIT " . $num * $limit . ", " . $limit;
```
Comment (sc979) [inline www/include/monitoring/downtime/listDowntime.php:187]: Are you sure that $num and $limit are sanitized ?

Comment (kduret) [reply www/include/monitoring/downtime/listDowntime.php:187]: it's not, I will manage it

File: www/include/monitoring/downtime/listDowntime.php [hunk 3]
```diff
@@ -179,18 +177,25 @@
         $centreon->user->access->getAccessGroupsString() . ") ";
 }
 $request .= (isset($search_service) && $search_service != "" ? "AND 1 = 0 " : "") .
```
Comment (callapa) [inline www/include/monitoring/downtime/listDowntime.php:179]: Is it possible to simplify or make these string concatenations more readable?

Comment (kduret) [reply www/include/monitoring/downtime/listDowntime.php:179]: I could, but not sure it is the purpose in this pull request

File: www/include/monitoring/downtime/listDowntime.php (hunk 1 above)
Comment (kduret) [reply www/include/monitoring/downtime/listDowntime.php:194]: no, I've tested it :)

**Review by callapa — APPROVED** [review]

File: www/include/monitoring/downtime/listDowntime.php (hunk 2 above)
Comment (kduret) [reply www/include/monitoring/downtime/listDowntime.php]: done

File: www/include/common/autoNumLimit.php [hunk 4]
```diff
@@ -38,9 +38,9 @@
 }
 
 if (isset($_POST["limit"]) && $_POST["limit"]) {
-    $limit = $_POST["limit"];
+    $limit = filter_input(INPUT_POST, 'limit', FILTER_VALIDATE_INT, array('options' => array('default' => 20)));
```
Comment (sc979) [inline www/include/common/autoNumLimit.php:41]: this will break almost 74 pages

**Review by sc979 — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**